src/dune_rules/merlin.mli

```diff
@@ -1,11 +1,43 @@
-(** Merlin rules *)
+(** This module ensures that one merlin configuration file is generated for each
+    stanza. Each of these configuration files is accompanied by a merlin exist
+    file. Each of these files contain a map from every module involved in the
+    stanza to a standard merlin configuration. The [Processed.t] type represents
+    the Merlin configuration as it will be marshalled to and from the
+    configuration files, while [Merlin.t] represents raw information coming from
+    the build system. *)
 open! Dune_engine
 
 open! Stdune
 open Import
 
+(** Merlin config file base name *)
+val merlin_file_name : string
+
+(** Merlin exist file base name *)
+val merlin_exist_name : string
+
+val make_lib_ident : Dune_file.Library.t -> string
```

Comment:
It would be helpful to leave a comment to the read about these merlin "ident" values.